pkg/openshift-routes/main.go

```diff
@@ -0,0 +1,30 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package platform contains the availability of the OpenShift Routes API.
+package openshift_routes
+
+// Platform holds the auto-detected platform type.
+type OpenShiftRoutesAvailability int
+
+const (
+	// Is not clear if OpenShift Routes are available.
+	Unknown OpenShiftRoutesAvailability = iota
```

Comment:
Why do we need both Unknown and NotAvailable states? 

If possible I would define only `Available` and `NotAvailable`

Reply:
Followed the approach suggested by @frzifus. Copied from the `platform` approach

Reply:
From my point of view it make sense to provide an unknown state too. Even it its the same as `NotAvailable` it provides the information that until then, the detection didnt run.